cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -266,6 +268,7 @@ def validate_and_update_from_json(cls, block, jsondict, user, filter_tabs=True):
                 did_validate = False
                 errors.append({'key': key, 'message': err_message, 'model': model})
 
+        cls.fill_teams_user_partitions_ids(block, filtered_dict)
```

Comment:
This is only called when advanced settings are updated, right?

Does that mean that if I enable `course_teams.content_groups_for_teams`, I will not be able to use team sets for a course until I edit its advanced settings?

Reply:
Exactly. I mentioned it before here: https://github.com/openedx/edx-platform/pull/33788#issuecomment-2010005998

> There is a scenario I'd like to discuss with the reviewer; consider this:
> - A new team is created in a course with the flag for this feature off
> - The flag is then turned on
> - The instructor tries to restrict content to teams
> - The instructor won't see any teams listed in the unit's content restrictions configurations
> 
> Why? The metadata used to link team sets to dynamic partitions is generated (if it isn't already) each time the course metadata is save; this can be triggered by saving any setting for the course in Studio. No, I wonder if this feature should handle this use case. Or should we recommend to turn the flag on and then create teams in the docs? 

Felipe reviewed this PR a few weeks ago and said [here](https://github.com/openedx/edx-platform/pull/33788#pullrequestreview-1964782756):

> the feature flag must be set before you activate teams. Or you need to disable teams/save/re-enable for the partition_ids to be generated. We should document that so that authors are not left wondering why the feature is not showing.

So, we decided to handle it in the feature documentation.

Reply:
If it makes it into the documentation, sounds good to me 👍🏻